### PR TITLE
ISPN-7398 Fix AbstractProtocolServerTest.testStartingWithoutTransport

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7398

### CODE REVIEW

File: server/core/src/test/java/org/infinispan/server/core/AbstractProtocolServerTest.java [hunk 1]
```diff
@@ -50,7 +52,7 @@ public void testStartingWithoutTransport() {
       b.startTransport(false);
       AbstractProtocolServer server = new MockProtocolServer();
       try {
-         server.start(b.build(), new DefaultCacheManager());
+         server.start(b.build(), new DefaultCacheManager(new ConfigurationBuilder().build()));
```
Comment (slaskawi) [inline server/core/src/test/java/org/infinispan/server/core/AbstractProtocolServerTest.java]: Why do we need this change? Are we deprecating the default constructor?

**Comment by wburns** [comment]
Looking...

**Comment by wburns** [comment]
I think we also need to clean up the `getCache(String, String)` method a bit

Maybe we should have it do

`if (DEFAULT_CACHE_NAME.equals(cacheName)) {
         log.deprecatedDefaultCache();
         return getCache();
      }
`

This way we also get the default cache exception thrown.

File: server/core/src/test/java/org/infinispan/server/core/AbstractProtocolServerTest.java (hunk 1 above)
Comment (tristantarrant) [inline server/core/src/test/java/org/infinispan/server/core/AbstractProtocolServerTest.java]: No, but the default constructor no longer provides a default cache, unless you pass it in.
The default constructor alone will give you a cache manager with no caches. You can use defineConfiguration() to provide some

**Comment by tristantarrant** [comment]
Hm, that would create an infinite loop, but yes we need a null handler there too. Let me fix it.

**Comment by tristantarrant** [comment]
@wburns fixed

**Comment by wburns** [comment]
Ah yes in the case when default cache name is set.  I was trying to trust
the of block instead of copying, but I guess copying is required :+1:

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request closed without merging**